services/slam/builtin/builtin.go

```diff
@@ -536,6 +539,18 @@ func NewBuiltIn(ctx context.Context, deps registry.Dependencies, config config.S
 		mapRate = *svcConfig.MapRateSec
 	}
 
+	var deleteProcessedData bool
+	if svcConfig.DeleteProcessedData == nil {
+		deleteProcessedData = !(offlineFlag)
+
+	} else {
+		deleteProcessedData = *svcConfig.DeleteProcessedData
+		if offlineFlag && deleteProcessedData {
```

Comment:
I think we don't have test coverage for this case. I suggest adding it as a unit test, i.e. check the command arguments produced by the slam service.